File: capabilities/business_domain/api/adapters.py
"""
业务领域 — 生态适配器 API 路由
"""
from fastapi import APIRouter, Query, Request

from jonex_core.common.exceptions import InvalidParameterError, JonexException
from jonex_core.common.i18n import translate
from jonex_core.common.response import error_response, success_response
from jonex_core.common.tenant import extract_tenant_id

from ..services import AdapterService

router = APIRouter()
_service = AdapterService()

VALID_ADAPTER_TYPES = {"dingtalk", "wechat_work", "feishu"}


def _validate_create_update(body: dict, *, require_name: bool = True) -> None:
    if require_name and not body.get("name"):
        raise InvalidParameterError(message=translate("err.adapter.name_required", fallback="适配器名称不能为空"), details={"field": "name"})  # 原消息
    adapter_type = body.get("adapter_type")
    if adapter_type and adapter_type not in VALID_ADAPTER_TYPES:
        raise InvalidParameterError(
            message=translate("err.adapter.unsupported_type", params={"adapter_type": adapter_type, "valid": ', '.join(sorted(VALID_ADAPTER_TYPES))}, fallback=f"不支持的适配器类型: {adapter_type}，必须是 {', '.join(sorted(VALID_ADAPTER_TYPES))} 之一"),  # 原消息
            details={"field": "adapter_type", "allowed": sorted(VALID_ADAPTER_TYPES)},
        )


@router.get("/adapters")
async def list_adapters(
    request: Request,
    offset: int = Query(0, ge=0),
    limit: int = Query(20, ge=1, le=100),
):
    tenant_id = extract_tenant_id(request)
    try:
        result = await _service.list(tenant_id, offset, limit)
        return success_response(data=result)
    except JonexException as e:
        return error_response(code=e.code, message=e.message, status_code=e.status_code, details=e.details)


@router.post("/adapters")
async def create_adapter(request: Request):
    body = await request.json()
    tenant_id = extract_tenant_id(request)
    _validate_create_update(body, require_name=True)
    try:
        result = await _service.create(tenant_id, body)
        return success_response(data=result, message="适配器已创建")
    except JonexException as e:
        return error_response(code=e.code, message=e.message, status_code=e.status_code, details=e.details)


@router.patch("/adapters/{adapter_id}")
async def update_adapter(adapter_id: str, request: Request):
    body = await request.json()
    tenant_id = extract_tenant_id(request)
    _validate_create_update(body, require_name=False)
    try:
        result = await _service.update(adapter_id, tenant_id, body)
        return success_response(data=result, message="适配器已更新")
    except JonexException as e:
        return error_response(code=e.code, message=e.message, status_code=e.status_code, details=e.details)


@router.post("/adapters/{adapter_id}/connect")
async def connect_adapter(adapter_id: str, request: Request):
    tenant_id = extract_tenant_id(request)
    try:
        result = await _service.connect(adapter_id, tenant_id)
        return success_response(data=result, message="适配器已连接")
    except JonexException as e:
        return error_response(code=e.code, message=e.message, status_code=e.status_code, details=e.details)


@router.post("/adapters/{adapter_id}/disconnect")
async def disconnect_adapter(adapter_id: str, request: Request):
    tenant_id = extract_tenant_id(request)
    try:
        result = await _service.disconnect(adapter_id, tenant_id)
        return success_response(data=result, message="适配器已断开")
    except JonexException as e:
        return error_response(code=e.code, message=e.message, status_code=e.status_code, details=e.details)

File: frontends/ecosystem-management/src/api/adapters.ts
import i18next from '@/locales/i18n'
import apiClient from './client'

interface ApiEnvelope<T> {
  success: boolean
  code?: number
  message?: string
  data?: T
}

export interface AdapterListResponse {
  items: AdapterItem[]
  total: number
  offset?: number
  limit?: number
}

export interface AdapterItem {
  id: string
  tenant_id: string
  name: string
  adapter_type: string
  config_json: Record<string, unknown>
  status: string
  created_at: string | null
  updated_at: string | null
}

export interface SaveAdapterPayload {
  name: string
  adapter_type: string
  config_json?: Record<string, unknown>
}

const ADAPTER_TYPE_LABELS: Record<string, string> = {
  dingtalk: 'adapter.platforms.dingtalk',
  wechat_work: 'adapter.platforms.wechat_work',
  feishu: 'adapter.platforms.feishu',
}

const STATUS_LABELS: Record<string, { label: string; color: string }> = {
  connected: { label: 'adapter.connected', color: 'green' },
  disconnected: { label: 'adapter.disconnected', color: 'default' },
  error: { label: 'adapter.error', color: 'red' },
}

export function getAdapterTypeLabel(type: string): string {
  return ADAPTER_TYPE_LABELS[type] || type
}

export function getAdapterStatusLabel(status: string): { label: string; color: string } {
  return STATUS_LABELS[status] || { label: status, color: 'default' }
}

export const ADAPTER_TYPE_OPTIONS: string[] = ['dingtalk', 'wechat_work', 'feishu']

function unwrapEnvelope<T>(payload: ApiEnvelope<T>): T {
  if (!payload?.success) {
    throw new Error(payload?.message || i18next.t('common.loadFailed'))
  }
  return payload.data as T
}

export async function listAdapters(
  offset = 0,
  limit = 100,
): Promise<AdapterListResponse> {
  const resp = await apiClient.get<ApiEnvelope<AdapterListResponse>>(
    '/api/v1/ecosystem/adapters',
    { params: { offset, limit } },
  )
  return unwrapEnvelope(resp.data)
}

export async function createAdapter(data: SaveAdapterPayload): Promise<AdapterItem> {
  const resp = await apiClient.post<ApiEnvelope<AdapterItem>>(
    '/api/v1/ecosystem/adapters',
    data,
  )
  return unwrapEnvelope(resp.data)
}

export async function updateAdapter(
  id: string,
  data: Partial<SaveAdapterPayload & { status: string }>,
): Promise<AdapterItem> {
  const resp = await apiClient.patch<ApiEnvelope<AdapterItem>>(
    `/api/v1/ecosystem/adapters/${id}`,
    data,
  )
  return unwrapEnvelope(resp.data)
}

export async function connectAdapter(id: string): Promise<AdapterItem> {
  const resp = await apiClient.post<ApiEnvelope<AdapterItem>>(
    `/api/v1/ecosystem/adapters/${id}/connect`,
  )
  return unwrapEnvelope(resp.data)
}

export async function disconnectAdapter(id: string): Promise<AdapterItem> {
  const resp = await apiClient.post<ApiEnvelope<AdapterItem>>(
    `/api/v1/ecosystem/adapters/${id}/disconnect`,
  )
  return unwrapEnvelope(resp.data)
}